app/src/main/java/com/guzelgimadieva/tasky/authorization/ui/login/LoginScreen.kt

```diff
@@ -54,38 +57,31 @@ import com.guzelgimadieva.tasky.core.theme.TaskyAppGreen
 import com.guzelgimadieva.tasky.core.theme.TaskyAppLightGray
 import com.guzelgimadieva.tasky.core.theme.TaskyAppPurple
 
-
-@Composable
-fun AuthorizationScreen(
-) {
-    val navController = rememberNavController()
-    NavHost(
-        navController = navController,
-        startDestination = AuthScreen.LOGIN.name,
-    ) {
-        composable(AuthScreen.LOGIN.name) {
-            LoginScreen(
-                onRegisterClick = {
-                    navController.navigate(AuthScreen.REGISTER.name)
-                }
-            )
-        }
-        composable(AuthScreen.REGISTER.name) {
-            RegisterScreen(
-                onBackClick = {
-                    navController.navigate(AuthScreen.LOGIN.name)
-                }
-            )
-        }
-    }
-}
-
 @Composable
 fun LoginScreen(
     onRegisterClick: () -> Unit,
 ) {
+
     val viewModel: LoginScreenViewModel = hiltViewModel()
     val loginState by viewModel.loginState.collectAsState()
+
+    val context = LocalContext.current
+    val refreshToken = stringResource(id = R.string.stored_refresh_token)
+    val userId = stringResource(id = R.string.stored_user_id)
+    LaunchedEffect(viewModel.refreshToken) {
+        viewModel.refreshToken.collect { token ->
+            context.getSharedPreferences(refreshToken, Context.MODE_PRIVATE)
+                .edit().putString(refreshToken, token).apply()
+            }
+        }
+    LaunchedEffect(viewModel.userId) {
+        viewModel.userId.collect { id ->
+            context.getSharedPreferences(userId, Context.MODE_PRIVATE)
+                .edit().putString(userId, id).apply()
+        }
+    }
```

Comment:
The UI shouldn't have to do with anything data related like loading stuff from preferences. On the one hand, you're forcing the app to use shared preferences here which will make it a pain to change it later on (e.g. to DataStore) and on the other hand, you'll include a lot of important logic in the UI. Logic that is directly in the UI is always the hardest to test, since you need to simulate a UI environment, which has to happen on an actual device (and is very slow and error prone).

Better create something like this:
```
data class AuthenticatedUser(
    val fullName: String,
    val email: String,
    val accessToken: String,
    val refreshToken: String
)
```
An abstraction which lets you easily replace the implementation with something else like DataStore:
```
interface UserPreferences {
    fun saveAuthenticatedUser(user: AuthenticatedUser)
    fun getAuthenticatedUser(): AuthenticatedUser?
    fun clearPreferences()
}
```
And only let the implementation contain the information that you're using shared preferences internally to handle prefs:
```
class UserPreferencesImpl(
    private val sharedPref: SharedPreferences
): UserPreferences {
    override fun saveAuthenticatedUser(user: AuthenticatedUser) {
        // Save JSON or fields of user in shared pref
    }

    override fun getAuthenticatedUser(): AuthenticatedUser? { ... }

    override fun clearPreferences() { ... }
}
```
In your Dagger module:
```
@Provides
@Singleton
fun provideUserPreferences(sharedPreferences: SharedPreferences): UserPreferences {
    return UserPreferencesImpl(sharedPreferences)
}
```
Then, for example in any ViewModel:
```
@HiltViewModel
class MyViewModel @Inject constructor(
    private val userPreferences: UserPreferences
): ViewModel() {
    // Use userPreferences
}
```